### PR TITLE
Issue/8172 site creation site type

### LINKED ISSUE #8172: Site Creation: Site Type Selection view
Add a view for selecting the new site's type.

From the design:
site_type_design

Needs Review: @nheagy

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #8172 

**To test:**
- Go to Switch Sites.
- Tap the "+" button to create a new site.
- Select "Create a WordPress.com site".
- Verify the select site type page displayed as looks like the design.

**Notes:**
- The design has a Back button. However, since this is the first page, there is nothing to go back to. So a Cancel button is displayed instead.
- The Help button will crash the app until #8199 is fixed.
- When a Site Type is selected, this temporary alert appears:
alert

**Implemented view:**

select_site_type

**Designed view:**

designed_view

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/SiteTypeTableViewCell.xib [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
+    </dependencies>
+    <objects>
+        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner"/>
+        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
+        <view contentMode="scaleToFill" id="iN0-l3-epB" customClass="SiteTypeTableViewCell" customModule="WordPress" customModuleProvider="target">
+            <rect key="frame" x="0.0" y="0.0" width="538" height="125"/>
+            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+            <subviews>
+                <stackView opaque="NO" contentMode="scaleToFill" alignment="center" spacing="10" translatesAutoresizingMaskIntoConstraints="NO" id="1EV-xM-yS2">
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteTypeTableViewCell.xib]: Love 😍 the stack views

File: Podfile.lock [hunk 2]
```diff
@@ -157,17 +157,11 @@ EXTERNAL SOURCES:
   Automattic-Tracks-iOS:
     :git: https://github.com/Automattic/Automattic-Tracks-iOS.git
     :tag: 0.2.1
-  WPMediaPicker:
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply Podfile.lock]: Honestly, I don't know. I merged develop & the podfile.lock changed, so I just checked it in as a matter of course. I'll merge again & see what happens.

File: WordPress/Classes/ViewRelated/NUX/SiteTypeTableViewCells.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+class InstructionTableViewCell: WPTableViewCell {
+
+    @IBOutlet weak var stepLabel: UILabel!
+    @IBOutlet weak var instr1Label: UILabel!
+    @IBOutlet weak var instr2Label: UILabel!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        assert(stepLabel != nil)
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/SiteTypeTableViewCells.swift]: I've never done that either, but I saw it in another class and thought it wasn't a bad idea. But I can remove it if it's quirky.

**Comment by ScoutHarris** [comment]
Hey @nheagy . Unless I should remove the awakeFromNib asserts, all done!

File: WordPress/Classes/ViewRelated/NUX/SiteTypeTableViewCells.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (nheagy) [reply WordPress/Classes/ViewRelated/NUX/SiteTypeTableViewCells.swift]: Nah, it's fine 😀

**Comment by nheagy** [comment]
Looks good!

:shipit:

**Pull request merged**